slugify/special.py

```diff
@@ -10,6 +11,7 @@ def add_uppercase_char(char_list):
         if upper_dict not in char_list and char != upper_dict[0]:
             char_list.insert(0, upper_dict)
         return char_list
+    raise RuntimeError("char_list must not be empty")
```

Comment:
Is this the only functional change in the PR? And if so, what makes it necessary?

Reply:
I believe so and if I recall, this was to avoid breaking compatibility with the typeshed def, https://github.com/python/typeshed/blob/3a376e7eaa279a8aac836acedad17b8eacb8c8e9/stubs/python-slugify/slugify/special.pyi#L3. I don't have opinions on where to break compatibility, if anywhere. Given they currently disagree, the only to prevent breaking compatibility in both would require type imprecision, which might be preferred if stability is the highest priority.

Reply:
There are definitely some behavioral changes in `test.py`, but this should have been the only behavioral change in the public API other than your other comment which is removed by fixing the annotation to the parse_args def as you suggested.